sycl/test/separate-compile/cpp_version_mismatch_test_2.cpp

```diff
@@ -0,0 +1,38 @@
+// >> ---- device compilation
+// RUN: %clangxx -std=c++14 -fsycl-device-only -Xclang -fsycl-int-header=sycl_ihdr_a.h %s -c -I %sycl_include
+
+// >> ---- code generation checks
+// RUN: FileCheck -input-file=sycl_ihdr_a.h %s
+// CHECK: #define STD_CPP_VERSION
```

Comment:
Is there a good reason to extract this into a macro instead of just putting it right into the conditional?

Reply:
You mean just call `getCppVersion(S)` in conditional? I don't see why that can't be done. @JoeyGIntel

Reply:
Actually, we ALREADY know if STD_CPP_VERSION >= 201401L, so emitting it int-header is foolish.  We should just omit this enitre set of code if that isn't the case .

Reply:
AND, it simplifies the patch a ton, you don't even NEED getCppVersion anymore, since you can just do:

if (LangOpts.CPlusPlus14) {
  os << "#if (__cplusplus <=201112L)\n";
  os << "    #error ...\n";
|

Reply:
I like this idea a lot!